psamm/datasource/native.py

```diff
@@ -74,7 +79,9 @@ def yaml_load(stream):
                            ' speed up loading the model files.')
 
     if _HAS_YAML_LIBRARY:
-        return yaml.load(stream, Loader=yaml.CSafeLoader)
+        loader = yaml.CSafeLoader
+        loader.add_constructor('tag:yaml.org,2002:float', float_constructor)
+        return yaml.load(stream, loader)
```

Comment:
This only works when `_HAS_YAML_LIBRARY` is true, I think. We need to also add the constructor when the `CSafeLoader` is not available and we fall back to using `yaml.safe_load()` (below). I think this would be easier if we always explicitly use the loader instead of using the `load()` or `safe_load()` wrapper. This way you can also add the constructor to the loader instance instead of the class. I think my comment on Trello explains this. To summarize: 1) Construct the loader explicitly (`loader = yaml.CSafeLoader(stream)`), 2) Add the constructor on the loader _instance_ (`loader.add_constructor(...)`), 3) Parse the stream and return the result (`return loader.get_data()`). When `_HAS_YAML_LIBRARY` is false we have to use the Python-based loader instead of `yaml.CSafeLoader`. I think it is called `yaml.SafeLoader`.